lntest/itest/lnd_rpc_middleware_interceptor_test.go

```diff
@@ -648,35 +652,66 @@ type middlewareHarness struct {
 
 // registerMiddleware creates a new middleware harness and sends the initial
 // register message to the RPC server.
-func registerMiddleware(t *testing.T, node *lntest.HarnessNode,
+func registerMiddleware(t *testing.T, node *node.HarnessNode,
 	registration *lnrpc.MiddlewareRegistration,
 	waitForRegister bool) *middlewareHarness {
 
-	ctxc, cancel := context.WithCancel(context.Background())
+	t.Helper()
 
-	middlewareStream, err := node.RegisterRPCMiddleware(ctxc)
-	require.NoError(t, err)
+	middlewareStream, cancel := node.RPC.RegisterRPCMiddleware()
 
-	err = middlewareStream.Send(&lnrpc.RPCMiddlewareResponse{
-		MiddlewareMessage: &lnrpc.RPCMiddlewareResponse_Register{
+	errChan := make(chan error)
+	go func() {
+		msg := &lnrpc.RPCMiddlewareResponse_Register{
 			Register: registration,
-		},
-	})
-	require.NoError(t, err)
+		}
+		err := middlewareStream.Send(&lnrpc.RPCMiddlewareResponse{
+			MiddlewareMessage: msg,
+		})
 
-	if waitForRegister {
-		// Wait for the registration complete message.
-		regCompleteMsg, err := middlewareStream.Recv()
-		require.NoError(t, err)
-		require.True(t, regCompleteMsg.GetRegComplete())
+		errChan <- err
+	}()
+
+	select {
+	case <-time.After(defaultTimeout):
+		require.Fail(t, "registerMiddleware send timeout")
+	case err := <-errChan:
+		require.NoError(t, err, "registerMiddleware send failed")
 	}
 
-	return &middlewareHarness{
+	mh := &middlewareHarness{
 		t:             t,
 		cancel:        cancel,
 		stream:        middlewareStream,
 		responsesChan: make(chan *lnrpc.RPCMessage),
 	}
+
+	if !waitForRegister {
+		return mh
+	}
+
+	// Wait for the registration complete message.
+	errChan = make(chan error)
+	msg := make(chan *lnrpc.RPCMiddlewareRequest)
+	go func() {
+		regCompleteMsg, err := middlewareStream.Recv()
+		if err != nil {
+			errChan <- err
+		} else {
+			msg <- regCompleteMsg
+		}
+	}()
+
+	select {
+	case <-time.After(defaultTimeout):
```

Comment:
style nit: add new lines between cases here and above.

Reply:
fixed!